### DISCUSSION TITLE
Plans for the new module federation version (enhanced) and esuild???

### DISCUSSION THREAD

**Discussion by eilensm** [nrwl/nx#26322] [Feature Requests]
With the upcoming support for esbuild from Zach Jackson concerning module federation (see https://x.com/ScriptedAlchemy/status/1794144831369035896), I would like to know if you have any plans to switch to the new module federation version (@module-federation/enhanced) and consider supporting esbuild?

This question is related to the older and closed issue: https://github.com/nrwl/nx/issues/17406

**Comment by jaysoo** [comment]
@eilensm Do you have a Angular or a React app? 

We are looking to migrate to `@module-federcation/enhanced`, but nothing for esbuild yet.

**Reply by eilensm** [reply]
We are building an Angular app. Saw some tutorials for using the new module federation version, but all of them are using React and rspack.

**Reply by soumyanildas** [reply]
I have done module federation 2.0 setup with React and Rspack. I am pretty sure we could just replace React with Angular. There's already a repo that showcases Rspack with Angular - https://github.com/jaysoo/ng-rspack.

If you want, we can have a discussion and brainstorm this together.

**Reply by eilensm** [reply]
My company is building an enterprise grade application, and as long as Angular does not have official support for rspack, it is not a viable option for us.

**Reply by Coly010** [reply]
@eilensm I'm curious as the process behind deciding that `rspack` is not a viable option because Angular does not support it officially.

Angular also does not officially support Module Federation, which changes the build output from webpack, impacting what the bundler is actually doing under the hood.

**Comment by eilensm** [comment]
It's simply trying to stick "close to the crowd" and keeping risks and their impact low. If rspack is getting more and more popular and we see more people adopting it, we surely will reconsider. And I'm always trying to keep up to date with what Zack J. is doing...
